simulator/mips/mips_instr.h

```diff
@@ -344,6 +348,23 @@ class MIPSInstr
         }
 
         uint64 get_v_dst() const { return v_dst; }
+        auto get_lwrl_mask() const
+        {
+            uint32 mask = 0x0;
+            uint32 i = 0;
+            for (i = 0; i < 4; ++i)                   // in this cycle we found byte in v_dst which is equal to v_imm
+                if ( v_imm == (v_dst >> i*8 & v_imm)) // and fill all previous bytes with zeros
```

Comment:
But again, you have to use "mem_addr" here instead of v_imm, right?

Reply:
I don't understand why should we use "mem_addr" instead of "v_imm"
.

Reply:
Example I provided above:

```asm
some_label:
; some data

la $at, some_label
addi $at, $at, 2 ; generating an address, $at % 4 == 2
lwl $at, 0($at) ; reading from unaligned address, mem_addr % 4 == 2, imm % 4 == 0
```